driver-core/src/main/com/mongodb/internal/operation/BsonDocumentWrapperHelper.java

```diff
@@ -25,7 +26,12 @@ final class BsonDocumentWrapperHelper {
 
     @SuppressWarnings("unchecked")
     static <T> List<T> toList(final BsonDocument result, final String fieldContainingWrappedArray) {
-        return ((BsonArrayWrapper<T>) result.getArray(fieldContainingWrappedArray)).getWrappedArray();
+        // BULK-TODO why does the expectation of this code fails?
```

Comment:
If the document doesn't contain a wrapped array why would you use this helper?

Reply:
I am simply constructing a new `CommandBatchCursor`. And for some reason, the `result` document, which I received from a `Connection.command` call that executed `bulkWrite`, contains a normal `BsonArray` in its `cursor.firstBatch` field, instead of containing a `BsonArrayWrapper` as the code in `BsonDocumentWrapperHelper.toList` expected. This does not seem right, given that in all other cases the expectation holds.

Reply:
My suggested edit in ClientBulkWriteOperation should fix this. That's what all the other cursor-initiating command do.

Reply:
Thank you!, it worked indeed.